include/scheduler.h

```diff
@@ -51,13 +51,16 @@ namespace detail {
 		graph_generator& ggen;
 		graph_serializer& gsrlzr;
 
-		std::queue<scheduler_event> events;
+		std::queue<scheduler_event> available_events;
+		std::queue<scheduler_event> in_flight_events;
```

Comment:
`in_flight_events` is only used inside `schedule()`, so it could be a local variable. This would make it more obvious that it is not synchronized by `events_mutex`.

Reply:
Good idea, done!